components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/util/HealthCheckConfigParser.java

```diff
@@ -0,0 +1,201 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.healthcheck.api.core.util;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.impl.builder.StAXOMBuilder;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.healthcheck.api.core.model.HealthCheckerConfig;
+import org.wso2.carbon.utils.CarbonUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLStreamException;
+
+import static org.wso2.carbon.healthcheck.api.core.Constants.HealthCheckConfig.DEFAULT_NAMESPACE;
+import static org.wso2.carbon.healthcheck.api.core.Constants.HealthCheckConfig.HEALTH_CHECKER;
+import static org.wso2.carbon.healthcheck.api.core.Constants.HealthCheckConfig.HEALTH_CHECKERS;
+import static org.wso2.carbon.healthcheck.api.core.Constants.HealthCheckConfig.NAME;
+import static org.wso2.carbon.healthcheck.api.core.Constants.HealthCheckConfig.ORDER;
+import static org.wso2.carbon.healthcheck.api.core.Constants.HealthCheckConfig.ENABLE;
+import static org.wso2.carbon.healthcheck.api.core.Constants.HealthCheckConfig.PROPERTY;
+import static org.wso2.carbon.healthcheck.api.core.Constants.HealthCheckConfig.PROPERTY_NAME;
+import static org.wso2.carbon.healthcheck.api.core.Constants.HealthCheckConfig.HEALTH_CHECK_CONFIG_FILE_NAME;
+
+/**
+ * Parses configurations in health-check-config.xml and build configuration objects.
+ */
+public class HealthCheckConfigParser {
+
+    private boolean isHealthCheckServiceEnabled = false;
+    private final static Log log = LogFactory.getLog(HealthCheckConfigParser.class);
+    private static HealthCheckConfigParser instance = new HealthCheckConfigParser();
+    private Map<String, HealthCheckerConfig> healthCheckerConfigMap = new HashMap<>();
+
+    public Map<String, HealthCheckerConfig> getHealthCheckerConfigMap() {
+
+        return healthCheckerConfigMap;
+    }
+
+    public static HealthCheckConfigParser getInstance() {
+
+        return instance;
+    }
+
+    private HealthCheckConfigParser() {
+
+        init();
+    }
+
+    private void init() {
+
+        try {
+            buildConfiguration();
+        } catch (FileNotFoundException e) {
+            // Configuration file is not mandatory. Hence log info and continue.
```

Comment:
If the config file is optional is it better to handle it from the logic, rather than letting the application to throw a file not found exception.

Reply:
FNF is thrown while reading the file so that we know file is not present.

Reply:
Use Path API
Files.exists(path)